### ISSUE TITLE
Port request for OwlCarousel

### ISSUE THREAD

**Issue by djzwerg** [backdrop-ops/contrib#450]
**Name of the module, theme, or layout**
OwlCarousel

**Link to the drupal.org module, theme, or layout**
https://www.drupal.org/project/owlcarousel

**Comment by rudy880719** [comment]
Hello, I want to migrate this module to backdrop, but I see that there are dependencies with the jquery_update and variable modules, these do not exist for backdrop, could you give me some guidance on this. It is also necessary to migrate these modules as well.

**Comment by djzwerg** [comment]
I think these modules won't bei needed in Backdrop CMS.

**Comment by djzwerg** [comment]
It looks like that OWL Carousel is dead. On their page they is noted that Tiny Slider should be used (https://github.com/ganlanyuan/tiny-slider). But there is bo D7 module for this.

**Comment by rudy880719** [comment]
@djzwerg I don't think so, Did you see the module page. https://www.drupal.org/project/owlcarousel, Why is dead?

**Comment by argiepiano** [comment]
`jquery_update` is not used in BAckdrop CMS. It was used in D7 to provide updates to jquery included in core. In Backdrop, we are currently using jquery 3, which is the latest major version.

As for the `variable` module - this would depend on how it's used in the D7 module. I see that `variable` provides metadata for D7 variables, allowing things such as access callbacks and tokens. Since Backdrop deprecated D7-like variables, then you would probably need to find a way to replicate those additional functionalities.

**Comment by djzwerg** [comment]
> @djzwerg I don't think so, Did you see the module page. https://www.drupal.org/project/owlcarousel, Why is dead?. I used it frequently
> 

The GitHub project is dead since six years (pls see https://github.com/OwlCarousel2/OwlCarousel2?tab=readme-ov-file). So there are no security patches or issue fixes anymore.

**Comment by rudy880719** [comment]
@djzwerg @argiepiano I came to this community to help. What other module do you think is a high priority to migrate?

**Comment by djzwerg** [comment]
I really love this module but I don't know how to maintain issues e.g. within the library. Maybe there is a chance to build a Module for Tiny Slider?

**Comment by rudy880719** [comment]
My apologies @djzwerg , You are right. I did't see the github page.

**Comment by rudy880719** [comment]
> I really love this module but I don't know how to maintain issues e.g. within the library. Maybe there is a chance to build a Module for Tiny Slider?

I will check it.. thanks

**Comment by rudy880719** [comment]
@djzwerg @argiepiano @laryn I have good news, I have built a version of[ Tiny Slider](https://github.com/rudy880719/tiny_slider) for backdrop. As we know OwlCarousel is dead, and this is an alternative,  I will appreciate if you review it and send me your comments and suggestions.

**Comment by rudy880719** [comment]
@djzwerg Hello, what is the step to publish this module in the community?

**Comment by argiepiano** [comment]
I've taken a quick look at your repo of Tiny Slider. 

- Avoid using functions that are not name-spaced in some way. Otherwise you risk a clash with others. For example, the function `defaultSettings()` should probably be `_tiny_slider_default_settings()`. My guess is that you copied this from a class method, where namespacing is not as crucial.  There are others...
- I noticed you are hard-coding the css classes in the template file. It would be advisable to set those classes in the template preprocess function, so that they can be expanded or overridden through another custom preprocess function. The way you are doing this, the only way to add or override those css classes is to override the whole template file.

**Comment by stpaultim** [comment]
@rudy880719 If we have not met before, welcome to the community and thanks for working on this module. 

I just tested it out and I love it. Very simple with basic functionality and it looks good. I have a non-profit site that needs a module like this one right now. We'll test it there as well.

**Comment by rudy880719** [comment]
@argiepiano Thank you for the code suggestions, and for the link to be a member of the community, in fact I'm not one yet, I appreciate your effort to review and provide corrections f or the code, it's very important to me those suggestions.

**Comment by rudy880719** [comment]
> @rudy880719 If we have not met before, welcome to the community and thanks for working on this module.
> 
> I just tested it out and I love it. Very simple with basic functionality and it looks good. I have a non-profit site that needs a module like this one right now. We'll test it there as well.
> 
> https://github.com/rudy880719/tiny_slider

@stpaultim Any time, thank you for the words, I appreciate it. I will continue working in this module, I want to add a formatter field to Entity Reference and Terms.

**Comment by stpaultim** [comment]
@rudy880719 

My colleagues looked at your module and like it better than the one we are using for a new site we are building right now. We'll likely use it on that project. Looking forward to an official release.

**Comment by stpaultim** [comment]
@rudy880719 We're testing this on a dev site in 2 places:

- http://fspco.dev.triplo.co
- http://fspco.dev.triplo.co/meet-our-board

We won't be able to use it when we launch unless it's on official contrib project. If you need any help moving it into contrib, please ask. I'm happy to help out in any way possible or to answer any questions you have.

**Comment by rudy880719** [comment]
Hello @stpaultim , I will need your help to move tiny slider and view timeline into contrib. Would you like to explain me how is the process. I have been busy, and I forgot it.
